### PR TITLE
Add basic installation instructions

### LINKED ISSUE #402: Installation Guide for the services
## Description

**As Josh I want to be able to**
find, understand and be able to install the Raiden services

**So that I can**

run the Raiden Services and earn fees

## Acceptance Criteria

- [x] Precise, exhaustive and easy to understand installation guide for the Raiden services (see Raiden Transport Readme for a good example https://github.com/raiden-network/raiden-transport)
- [ ] 2 Trustlines devs should be able to install the Raiden services without any open questions
- [x] Installation guide should be included in the Readme of the services repo and in the services spec , see https://raiden-network-specification.readthedocs.io/en/latest/pathfinding_service.html
- [ ] Additionally, we need an explanation on how to maintain the services, e.g. how to change and claim services fees and how to monitor the services  
- [x] Add a script to register a service in the `ServiceRegistry` (#448 )

### PR DESCRIPTION
**Pull request by palango**
Mostly copied from the raiden-transport setup.

Fixes #402

### CODE REVIEW

File: package/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - /data/state:/state
+    - /data/keystore:/keystore
+
+services:
+  pfs:
+    << : *defaults
+    command: ["python3", "-m", "pathfinding_service.cli", "--enable-debug"]
+    labels:
+      - "traefik.enable=true"
+      - "traefik.frontend.rule=Host: pfs-ropsten-with-fee.services-dev.raiden.network"
```
Comment (palango) [inline package/docker-compose.yml]: @ulope Can I somehow make this dynamic based on some variable in the `.env`  file?

**Comment by palango** [comment]
@ulope I need some help with writing how to setup the ethereum node. I guess people shouldn't run it on the same server. How would you handle that?

**Comment by ulope** [comment]
> I need some help with writing how to setup the ethereum node. I guess people shouldn't run it on the same server. How would you handle that?

I'd just specify that as a requirement. 
e.g. `A synced Ethereum node, that is reachable from the server where the services are installed, is required. Setting one up is outside of the scope of this document, please refer to <some link to eth node setup instructions>.`

File: package/README.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+After a successful deployment the following ports will be in use:
+
+- 80 - HTTP
+  - Redirects to HTTPS
+  - Let's Encrypt HTTP challenge for certificate provisioning
+- 443 - HTTPS
+  - Synapse web and API client access
+  - Metrics export (IP restricted, see below)
+ 
+TODO: check
```
Comment (Dominik1999) [inline package/README.md]: what do you want to check here?

File: package/README.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+   Examples:
+   - raidenservices.mydomain.tld
+   - raidenservices-mydomain.tld
+   - mydomain-raidenservices.tld
+
+1. Configure `A` (and optionally `AAAA`) DNS records for the domain pointing to the servers IP address(es)
+1. Configure a `CNAME` DNS record for `*.<domain>` pointing back to `<domain>`
+
+TODO: check
```
Comment (Dominik1999) [inline package/README.md]: what do you want to check here?

File: package/README.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+### Installing
```
Comment (Dominik1999) [inline package/README.md]: nice installation guide!

File: package/README.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+1. Clone the [current release version of this repository](https://github.com/raiden-network/raiden-services)
+   to a suitable location on the server:
+
+   ```shell
+   git clone https://github.com/raiden-network/raiden-services.git
+   ```
+1. Copy the keystore file to be used by the services into the directory `./data/keystore/`
```
Comment (Dominik1999) [inline package/README.md]: it should be `2.` and not `1.` again, right?

Comment (Dominik1999) [inline package/README.md]: I have problems with permissions using that folder. It works but it is not obvious to copy with sudo and change permissions after that. Is that something we can change?

File: README.md [hunk 6]
```diff
@@ -22,11 +23,14 @@ For more technical details see:
 - [Monitoring Service specification](https://raiden-network-specification.readthedocs.io/en/latest/monitoring_service.html)
 - [Pathfinding Service specification](https://raiden-network-specification.readthedocs.io/en/latest/pathfinding_service.html)
 
-## Getting started
+## Service Providers
+
+To become a service provider follow the instructions in the [service provider README](https://github.com/raiden-network/raiden-services/blob/master/package/README.md).
```
Comment (Dominik1999) [inline README.md]: do you link to the very same file that you are writing at the moment?

File: package/.env.template [hunk 7]
```diff
@@ -0,0 +1,33 @@
+## This is a template for a docker-compose .env file
+
+## The FQDN name this server will be reachable on
+SERVER_NAME=somedomain.example.com
+
+## Email addres to use when requesting LetsEncrypt certificates.
```
Comment (Dominik1999) [inline package/.env.template]: it is "address", right?

Comment (Dominik1999) [reply package/.env.template]: Can you link to LetsEncrypt certificates? I don't know what this is

File: package/.env.template [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+## This will only be used by letsencrypt to notify about pending certificate
```
Comment (Dominik1999) [inline package/.env.template]: letsencrypt looks different to the above LetsEncrypt

File: package/.env.template [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+## expiration (which normally shouldn't happen).
+LETSENCRYPT_EMAIL=email@example.com
+
+## Data dir location. Optional, defaults to ./data in the checkout directory.
+#DATA_DIR=/data
+
+## Services settings
+## the keystore file, has to be located in ${DATA_DIR}/keystore
```
Comment (Dominik1999) [inline package/.env.template]: The capitalized?

File: package/README.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
+# Raiden Services
```
Comment (Dominik1999) [inline package/README.md]: This looks honestly very nice! Thanks for creating that!

File: package/README.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+- [Changelog](#changelog)
+
+## Overview
+
+The Raiden Network uses pathfinding and monitoring services to increase usability.
+To ensure reliability, availability and neutrality it is desirable that those servers are
+being operated by multiple independent entities.
+
+Therefore we provide this repository which allows easy setup of these services.
+It uses docker and docker-compose for easy installation and upgrades.
```
Comment (Dominik1999) [inline package/README.md]: and Traefik or do we not need to emphasize that?

File: package/README.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+
+### Used software
+
+- docker
```
Comment (Dominik1999) [inline package/README.md]: do we need to add a version number?

File: package/README.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+          |                        |
++---------v---------+    +---------v---------+
+|                   |    |                   |
+|     Pathfinding   |    |    Monitoring     |
+|                   |    |                   |
++---------+---------+    +---------+---------+
+```
+
+
+We use Traefik as a reverse proxy and also utilize it's capability of automatically provisiong
```
Comment (Dominik1999) [inline package/README.md]: there is a "." missing?

Comment (Dominik1999) [reply package/README.md]: and at least I don't know the word "provisiong"?

File: package/README.md [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+Let's Encrypt TLS certificates.
```
Comment (Dominik1999) [inline package/README.md]: ah the sentence continues here, but LetsEncrypt is again different to the previous ones

File: package/README.md [hunk 15]
```diff
@@ -0,0 +1,178 @@
...
+
+### Network
+
+After a successful deployment the following ports will be in use:
+
+- 80 - HTTP
+  - Redirects to HTTPS
+  - Let's Encrypt HTTP challenge for certificate provisioning
```
Comment (Dominik1999) [inline package/README.md]: LetsEncrypt ?1

File: package/README.md [hunk 16]
```diff
@@ -0,0 +1,178 @@
...
+ 
+TODO: check
+
+## Requirements
+
+### Hardware
+
+Minumum recommended for a production setup:
+
+- 2 GiB Ram
```
Comment (Dominik1999) [inline package/README.md]: is "GiB" the correct abbreviation or is it "GB"?

Comment (Dominik1999) [reply package/README.md]: and it is RAM, right?

File: package/README.md [hunk 17]
```diff
@@ -0,0 +1,178 @@
...
+- 2 Cores
+- 10 GiB SSD
+
+### Software
```
Comment (Dominik1999) [inline package/README.md]: You have another chapter software above? That seems to double. Can we remove one?

File: package/README.md [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+
+- Docker >= 17.12
+- docker-compose >= 1.21.0
```
Comment (Dominik1999) [inline package/README.md]: is Traefik also a software?

File: package/README.md [hunk 19]
```diff
@@ -0,0 +1,178 @@
...
+
+### Other
+
+- A domain (or subdomain) for exclusive use by the services
```
Comment (Dominik1999) [inline package/README.md]: per service or for the two services?

File: package/README.md (hunk 5 above)
Comment (palango) [reply package/README.md]: That gets rendered properly by the markdown parser.

File: package/README.md (hunk 15 above)
Comment (palango) [reply package/README.md]: That's a service that provides SSL certificates for free, something that was quite expensive before. https://letsencrypt.org/

File: package/README.md (hunk 17 above)
Comment (palango) [reply package/README.md]: Where, I only see one?

File: package/README.md (hunk 18 above)
Comment (palango) [reply package/README.md]: Yes, but it's downloaded through docker, so we don't need to listen it here.

File: package/README.md (hunk 19 above)
Comment (palango) [reply package/README.md]: one domain is enough, we can create subdomains through traefik

File: package/README.md (hunk 5 above)
Comment (palango) [reply package/README.md]: It's strange that your keystore file was written by root. But I'll add a hint that the docker container needs permissions to read that file.

Comment (Dominik1999) [reply package/README.md]: yes but it also happened the second time, so maybe that is a ubuntu problem

File: package/README.md (hunk 17 above)
Comment (Dominik1999) [reply package/README.md]: ok you are right

**Pull request merged**